core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -150,6 +173,76 @@ object DumpLogSegments {
     }
   }
 
+  private def dumpTimeIndex(file: File,
+                            indexSanityOnly: Boolean,
+                            verifyOnly: Boolean,
+                            timeIndexDumpErrors: TimeIndexDumpErrors,
+                            maxMessageSize: Int) {
+    val startOffset = file.getName().split("\\.")(0).toLong
+    val logFile = new File(file.getAbsoluteFile.getParent, file.getName.split("\\.")(0) + Log.LogFileSuffix)
+    val messageSet = new FileMessageSet(logFile, false)
+    val indexFile = new File(file.getAbsoluteFile.getParent, file.getName.split("\\.")(0) + Log.IndexFileSuffix)
+    val index = new OffsetIndex(indexFile, baseOffset = startOffset)
+    val timeIndex = new TimeIndex(file, baseOffset = startOffset)
```

Comment:
When I ran the following command, I got an exception. 

bin/kafka-run-class.sh kafka.tools.DumpLogSegments --files 00000000000000113931.timeindex 
Dumping /Users/junrao/Downloads/00000000000000113931.timeindex
Exception in thread "main" java.lang.IllegalArgumentException: Invalid max index size: -1
	at kafka.log.AbstractIndex.<init>(AbstractIndex.scala:54)
	at kafka.log.OffsetIndex.<init>(OffsetIndex.scala:51)
	at kafka.tools.DumpLogSegments$.kafka$tools$DumpLogSegments$$dumpTimeIndex(DumpLogSegments.scala:185)
	at kafka.tools.DumpLogSegments$$anonfun$main$1.apply(DumpLogSegments.scala:101)
	at kafka.tools.DumpLogSegments$$anonfun$main$1.apply(DumpLogSegments.scala:91)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:108)
	at kafka.tools.DumpLogSegments$.main(DumpLogSegments.scala:91)
	at kafka.tools.DumpLogSegments.main(DumpLogSegments.scala)